components/eam/src/physics/cam/modal_aer_opt.F90

```diff
@@ -1265,17 +1301,30 @@ subroutine modal_aero_lw(list_idx, state, pbuf, tauxar)
    call rad_cnst_get_info(list_idx, nmodes=nmodes)
 
    if (list_idx == 0) then
-      ! water uptake and wet radius for the climate list has already been calculated
-      call pbuf_get_field(pbuf, dgnumwet_idx, dgnumwet_m)
-      call pbuf_get_field(pbuf, qaerwat_idx,  qaerwat_m)
+      ! Calculate aerosol size distribution parameters and aerosol water uptake
+      ! For prescribed aerosol codes
+      if (clim_modal_aero .and. .not. prog_modal_aero) then
+         ! diagnostic aerosol size calculations
+         call modal_aero_calcsize_diag(state, pbuf, list_idx, dgnumdry_m)
+         if (present(clear_rh)) then
+            ! clear_rh provides alternate estimate non-cloudy relative humidity
+            call modal_aero_wateruptake_dr(state, pbuf, list_idx, dgnumdry_m, dgnumwet_m, &
+              qaerwat_m, wetdens_m, clear_rh_in=clear_rh)
+         else
+            call modal_aero_wateruptake_dr(state, pbuf, list_idx, dgnumdry_m, dgnumwet_m, &
+              qaerwat_m, wetdens_m)
+         endif
+      else
+         !For prognostic aerosols
+         call modal_aero_calcsize_sub(state, dt, pbuf,list_idx_in=list_idx, dgnumdry_m=dgnumdry_m)
+         call modal_aero_wateruptake_dr(state, pbuf, list_idx, dgnumdry_m, dgnumwet_m, &
```

Comment:
Don't we need logic for when clear_rh is present here?

Reply:
I will clean this up in my next BFB PR. Since MMF was not using prognostic aerosols, I didn't include this logic for `clear_rh`

Reply:
Regardless the answer is "no". The "presence" of an optional parameter like `clear_rh` would be passed to a subroutine. So actually I think this whole block can be cleaned up by just passing `clear_rh_in=clear_h` in both cases. Nothing wrong with being explicit though probably, so long as the clear_rh present case gets fixed.